src/components/now-playing-drawer/Scrubber.tsx

```diff
@@ -0,0 +1,284 @@
+    root: {
+      flexDirection: 'row',
+      alignItems: 'center'
+    },
+    timestamp: {
+      color: themeColors.neutral,
+      fontSize: 12,
+      flexShrink: 1
+    },
+    slider: {
+      height: 4,
+      marginLeft: 16,
+      marginRight: 16,
+      flexGrow: 1
+    },
+    rail: {
+      backgroundColor: themeColors.neutralLight8,
+      borderRadius: 2,
+      borderColor: themeColors.neutralLight8,
+      height: 4,
+      overflow: 'hidden'
+    },
+    tracker: {
+      // The tracker must be the full width of the rail
+      // so that it can have rounded edges. It animates
+      // by sliding (translateX) instead of by growing.
+      flexGrow: 1,
+      backgroundColor: themeColors.neutral,
+      borderRadius: 2,
+      borderColor: themeColors.neutralLight8,
+      height: 4
+    },
+    handleContainer: {
+      top: -6,
+      position: 'absolute'
+    },
+    handle: {
+      marginLeft: -8,
+      height: 16,
+      width: 16,
+      borderRadius: 8,
+      backgroundColor: themeColors.staticWhite,
+      // Note: React-native-shadow-2 seems to lose fidelity
+      // when styling such a small shadow.
+      // TODO: Revisit this, but as of writing, these values
+      // are fairly good on android/ios
+      shadowColor: 'rgb(133,129,153)',
+      shadowOffset: {
+        width: 0,
+        height: 1
+      },
+      shadowOpacity: 0.5,
+      shadowRadius: 2,
+      elevation: 3
+    }
+  })
+
+const attachToDx = (animation: Animated.Value, newValue: number) => (
+  e: GestureResponderEvent
+) => {
+  Animated.event(
+    [
+      null,
+      {
+        dy: animation
+      }
+    ],
+    { useNativeDriver: false }
+  )(e, { dy: newValue })
+}
+
+type ScrubberProps = {
+  mediaKey: string
+}
+
+// Memoize because the Scrubber is sensitive to parent rerenders
+// due to its gesture handler use.
+// useSelectorWeb (used to connect the now playing drawer) induces
+// rerenders that normally should be harmless.
+export const Scrubber = memo(({ mediaKey }: ScrubberProps) => {
+  const styles = useThemedStyles(createStyles)
+  const { primaryLight2, primaryDark2 } = useThemeColors()
+
+  // Animation to translate the handle and tracker
+  const translationAnim = useRef(new Animated.Value(0)).current
+  // Scale animation for the handle
+  const {
+    scale: handleScale,
+    handlePressIn: handlePressHandleIn,
+    handlePressOut: handlePressHandleOut
+  } = usePressScaleAnimation(HANDLE_GROW_SCALE)
+
+  const railRef = React.createRef<View>()
+  // The rail component's width
+  const [railWidth, setRailWidth] = useState(0)
+  // The rail component's distance from the left
+  const [railPageX, setRailPageX] = useState(0)
+  // The position of the drag-handle
+  const [handlePosition, setHandlePosition] = useState(0)
+
+  useEffect(() => {
+    if (railRef.current) {
+      railRef.current.measure((x, y, width, height, pageX, pageY) => {
+        setRailPageX(pageX)
+      })
+    }
+  }, [railRef])
```

Comment:
i think we want to do railRef.current, cause not use if railRef itself will ever change instances?

Reply:
I don't think it matters here b/c the "measure" only needs to run at mount since the size of the thing doesn't change. unf we can't use onLayout b/c that doesn't give us pageX